src/pyggdrasil/distances/_scphylo_wrapper.py

```diff
@@ -32,7 +36,12 @@ def calculate(self, /, tree1: anytree.Node, tree2: anytree.Node) -> float:
 
         df1 = utils.tree_to_dataframe(tree1)
         df2 = utils.tree_to_dataframe(tree2)
-        return scphylo.tl.ad(df1, df2)
+        try:
```

Comment:
I guess raising the error is better:

```
try:
    return scphylo.tl.ad(df1, df2)
except ZeroDivisionError:
    raise ZeroDivisionError("Error known to happen when no pairs of mutations exist in input tree. Consider <the name of the improved metric>") 
```

In particular, NaNs could be propagated further down and then e.g., averages or some plots will have some unexpected behaviour.

Reply:
Good point! What do you think about something like this?
 
```
class AncestorDescendantSimilarity(interface.TreeSimilarity):
    def __init__(self, raise_error: bool) -> None:
        self._raise_error = raise_error
    
    ...

        try:
            distance = scphylo.tl.ad(df1, df2)
        except ZeroDivisionError:
            if self._raise_error:
               raise ValueError("Error known to happen when no pairs of mutations exist in input tree.")
            else:
                return jnp.nan
```

Reply:
Yes - that surely works. 

Actually, that is just what Farid just did in this [commit](https://github.com/faridrashidi/scphylo-tools/commit/6f3c65cd41c80926eefa700735f85a71bc42ca0c) 

Perhaps we should update our fork  - Or is it easier to just do this ourselves? 

But that still leaves the question of what the snakemake workflow is supposed to do when calculating distances of star trees. The two options seem to be:

1) excluding them from the workflow - i.e. not generating the files.  - implies a somewhat cumbersome output filepath generation for all star tree / AD experiments
2) generate `nan` and empty plots

I am inclined to the first option.

Reply:
As I haven't touched the fork for quite some time, I worry that it may have diverged by quite a lot and trying to update it would lead to all sorts of compatibility issues :sweat_smile: 

As your solution below works we can drop modifying our A-D distance as well :slightly_smiling_face: